znnl/models/huggingface_flax_model.py

```diff
@@ -0,0 +1,195 @@
+If you use this module please cite us with:
+
+Summary
+-------
+Module for using a Flax model from Hugging Face in ZnNL.
+"""
+
+import logging
+from typing import Any, Callable, List, Sequence, Union
+
+import jax.numpy as np
+import optax
+from flax import linen as nn
+from flax.training.train_state import TrainState
+from transformers import FlaxPreTrainedModel
+
+from znnl.models.jax_model import JaxModel
+from znnl.optimizers.trace_optimizer import TraceOptimizer
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceFlaxModel(JaxModel):
+    """
+    Class for a Hugging Face (HF) flax model.
+    """
+
+    def __init__(
+        self,
+        pre_built_model: FlaxPreTrainedModel,
+        optimizer: Callable,
+        batch_size: int = 10,
+        trace_axes: Union[int, Sequence[int]] = (-1,),
+        store_on_device: bool = True,
+    ):
+        """
+        Constructor of a HF flax model.
+
+        Parameters
+        ----------
+        pre_built_model : FlaxPreTrainedModel or subclass
+                Pre-built model from Hugging Face.
+        optimizer : Callable
+                optimizer to use in the training. OpTax is used by default and
+                cross-compatibility is not assured.
+        input_shape : tuple
+                Shape of the NN input.
+        batch_size : int
+                Size of batch to use in the NTk calculation.
+        trace_axes : Union[int, Sequence[int]]
+                Tracing over axes of the NTK.
+                The default value is trace_axes(-1,), which reduces the NTK to a tensor
+                of rank 2.
+                For a full NTK set trace_axes=().
+        store_on_device : bool, default True
+                Whether to store the NTK on the device or not.
+                This should be set False for large NTKs that do not fit in GPU memory.
+        """
+        logger.info(
+            "Flax models have occasionally experienced memory allocation issues on "
+            "GPU. This is an ongoing bug that we are striving to fix soon."
+        )
+
+        self.apply_fn = pre_built_model.__call__
+        self.module = pre_built_model.module
+        # self.config = pre_built_model.config
+
+        # Save input parameters, call self.init_model
+        super().__init__(
+            pre_built_model=pre_built_model,
+            optimizer=optimizer,
+            trace_axes=trace_axes,
+            ntk_batch_size=batch_size,
+            store_on_device=store_on_device,
+        )
+
+    def _ntk_apply_fn(self, params: dict, inputs: np.ndarray):
+        """
+        Return an NTK capable apply function.
+
+        Parameters
+        ----------
+        params : dict
+                Network parameters to use in the calculation.
+        x : np.ndarray
+                Data on which to apply the network
+
+        Returns
+        -------
+        Acts on the data with the model architecture and parameter set.
+        """
+        # out = self.apply_fn(
```

Comment:
Can you please remove the commented out code?

Reply:
Done in [339a436](https://github.com/zincware/ZnNL/pull/105/commits/339a436bb58d2c09694cb6db2aeecc47887f1cdf)